operators/src/pro/machine_learning/onnx.rs

```diff
@@ -0,0 +1,473 @@
+                    for y in 0..height {
+                        for x in 0..width {
+                            let pixel_index = y * width + x;
+                            let pixel_value = tile
+                                .get_at_grid_index(GridIdx2D::from([y as isize, x as isize]))?
+                                .unwrap_or(f32::NAN); // TODO: skip the pixel entirely instead?
+                            pixels[pixel_index][tile_index] = pixel_value;
+                        }
+                    }
+                }
+
+                let pixels = pixels.into_iter().flatten().collect::<Vec<f32>>();
+                let rows = width * height;
+                let cols = num_bands;
+
+                let samples = Array2::from_shape_vec((rows, cols), pixels).expect(
+                    "Array2 should be valid because it is created from a Vec with the correct size",
+                );
+
+                let input_name = &session.inputs[0].name;
+
+                let outputs = session
+                    .run(ort::inputs![input_name => samples].context(error::Onnx)?)
+                    .context(error::Onnx)?;
+
+                let predictions = outputs["output_label"]
+                    .try_extract_tensor::<i64>()
+                    .context(error::Onnx)?;
+                let predictions = predictions.into_owned().into_raw_vec();
+
+                Ok(RasterTile2D::new(
+                    time,
+                    tile_position,
+                    0,
+                    global_geo_transform,
+                    Grid::new([width, height].into(), predictions)?.into(),
+                    cache_hint,
+                ))
+            });
+
+        Ok(stream.boxed())
+    }
+
+    fn raster_result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use geoengine_datatypes::{
+        primitives::{CacheHint, SpatialPartition2D, SpatialResolution, TimeInterval},
+        raster::{GridShape, RenameBands, TilesEqualIgnoringCacheHint},
+        spatial_reference::SpatialReference,
+        util::test::TestDefault,
+    };
+    use ndarray::{arr2, Array1, Array2};
+
+    use crate::{
+        engine::{MockExecutionContext, MockQueryContext, MultipleRasterSources},
+        mock::{MockRasterSource, MockRasterSourceParams},
+        processing::{RasterStacker, RasterStackerParams},
+    };
+
+    use super::*;
+
+    #[test]
+    fn tract() {
+        use tract_onnx::prelude::*;
+
+        let model = tract_onnx::onnx()
+            // load the model
+            .model_for_path(test_data!("pro/ml/onnx/test_no_zipmap.onnx"))
+            .unwrap()
+            .into_runnable()
+            .unwrap();
+
+        let array = arr2(&[[0.1f32, 0.2], [0.2, 0.3], [0.2, 0.2], [0.3, 0.1]]);
+
+        let tensor: Tensor = array.into();
+
+        // run the model on the input
+
+        // TODO: fix the input because the prediction is wrong(?)
```

Comment:
is this still the case?

Reply:
yes. If we make a final decision on ort than we can just remove this test. I still can't get tract to work.